herbert_core/utilities/misc/try_to_create_folder.m

```diff
@@ -0,0 +1,38 @@
+function [success,folder_path,mess]=try_to_create_folder(location,folder_name)
+% The function tries to creates folder in specified directory, checking for
+% it existence first. If folder exists, the routine checks that this folder is
+% accessible for writing. 
+% 
+% Inputs:
+% location       -- the topmost location, where the folder should be created
+% folder_name    -- the name of the folder to create
+%
+% Returns:
+% success     -- true if the folder is created and available for writing or
+%                exists and is available for writing.
+% folder_path -- the path to the created folder.
+% mess        -- empty if success or reason for the issue if it was not.
+%
+folder_path = [location,filesep,folder_name];
+
+ic = 0;
+if ~exist(folder_path,'dir')
+    [success, mess] = mkdir(folder_path);
+    while ~(exist(folder_path,'dir') || ic<3)
+        [success, mess] = mkdir(folder_path);
+        ic = ic + 1;
+        pause(0.1);
+    end
+else
```

Comment:
what is this else branch doing? if the folder doesn't exist it tries to create it; if it does exist it tried to create randomly named path and immediately delete it

Reply:
yes -- its the part of the code, which creates folders for configuration (I've reused existing piece of code for tmp folder.)
The issue is to ensure that the folder have correct write access. 
Apparently, creating something is simplest and most reliable way of checking write access on any system.

Reply:
delay and repetitive attempts to create come from Windows, where in some configurations something (looks like antivirus (or virus?) temporary locks some folders.